CIP-0XXX/README.md

```diff
@@ -0,0 +1,561 @@
+- Verify the KES signature of the message body with the KES verification key from the operational certificate.
+- Compute the SPO pool id by hashing the cold verification key from the operational certificate.
+- Make sure that this pool id is part of the stake distribution (the network layer will need to have access to this information).
+
+If any of these step fails, the message is considered as invalid, which is a protocol violation.
+
+> [!WARNING]  
+> We also probably need to make sure that the KES key used to sign is from the latest rotation:
+> - either the last seen opcert number in the block headers of the chain.
+> - or the last seen opcert number from a previous message diffused.
+> 
+> If the opcert number received is strictly lower than the previous one which has been seen, it > should be considered as a protocol violation.
+
+#### Cost of authentication
+
+> [!NOTE]
+> Computations are based on a **2 ms** verification delay per message on a CPU core.
+
+For a total of **3,100** Cardano SPOs, there will be **1,550** messages sent per round of signature: 
+
+- the number of messages received by a node which need to be verified is:
+
+| Send period | Messages sent |
+| ----------- | ------------- |
+| 1 min       | 64 M/month    |
+| 2 min       | 32 M/month    |
+| 5 min       | 13 M/month    |
+| 10 min      | 7 M/month     |
+
+- the extra CPU time for verification for a Cardano full node is:
+
+| Send period | CPU core usage |
+| ----------- | -------------- |
+| 1 min       | 5%             |
+| 2 min       | 2.5%           |
+| 5 min       | 1.0%           |
+| 10 min      | 0.5%           |
+
+### Possible attacks
+
+#### Sybil attack
+
+In this attack, a malicious sender would attempt to create multiple identities impersonating SPOs. This attack is completely mitigated by the [Message authentication mechanism](#Message-authentication-mechanism) as only active SPO on the Cardano chain can be authenticated and send messages. This would be considered as a protocol violation and the malicous peer would be disconnected.
+
+#### Equivocation
+
+In this attack, a malicious SPO would send different messages to different peers. This attack needs to be handled by the receiver of the message as the network layer does not verify the content of the message body by design.
+
+In the specific case of Mithril, the individual signature is unique so there will be two cases:
+- the message embeds a valid signature and it will be accepted by the receiving Mithril aggregator.
+- the message embeds an invalid signature and it will be rejected by the receiving Mithril aggregator.
+
+#### DoS attack
+
+In this attack, a malicous SPO would try to flood the network by sending many messages at once. In that case, the network layer could detect that the throughput of messages originating from a SPO is above a threshold and consider it as a protocol violation, thus disconnecting the malicous peer.
```

Comment:
> messages originating from a SPO is above a threshold

This implies that the protocol should be configured with thresholds. Would be great if we can mention this somewhere else

Reply:
In a pull-based protocol this would already be a protocol violation. If a peer ask for N signatures and receives N+1 then that's enough to be a protocol violation. Also the way miniprotocols are implemented they allow to set a maximum message size that would also be a threshold. We can make both mentions here

Reply:
I added mentions in the document